### PR TITLE
Configure checkstyle in the project, add properties file and implement DriverManager singleton pattern

### PR DESCRIPTION
**Pull request by danielaSantaCruz**

### CODE REVIEW

File: src/main/java/core/DriverManager.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Gets the driver of the given driver.
+     *
+     * @param browserDriver String with the type of driver.
+     * @return a WebDriver.
+     */
+    public static WebDriver getDriver(final String browserDriver) {
+        browser = browserDriver;
+        if (driver == null) {
+            if (browser.equalsIgnoreCase("chrome")) {
```
Comment (jpgonzales27) [inline src/main/java/core/DriverManager.java]: Maybe, you can use the default browser what are included in the Bonigarcia dependency

File: src/main/java/core/DriverManager.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Manages and initializes the browser drivers.
+ */
+public class DriverManager {
+
+    public static WebDriver driver = null;
+    public static String browser;
+
+    private DriverManager() {
```
Comment (jpgonzales27) [inline src/main/java/core/DriverManager.java]: I think this singleton should be modified and return only the instance of driver, like we saw in the class today

File: src/main/java/salesforce/ui/pages/LegalEntityPage.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * Gets the success message.
+     *
+     * @return a String with the message.
+     */
+    public String getUserSuccessMessage() {
+        successMessage.click();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
```
Comment (jpgonzales27) [inline src/main/java/salesforce/ui/pages/LegalEntityPage.java]: Maybe, this  time will be defined in other singleton pattern for the waits time

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+group 'org.example'
+version '1.0-SNAPSHOT'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+//    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
```
Comment (edsonarios) [inline build.gradle]: These comments should be deleted

File: src/main/java/salesforce/ui/pages/LoginPage.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Clicks the login button.
+     *
+     * @return A homePage.
+     */
+    public HomePage clickLoginBtn() {
+        loginBtn.click();
+        return new HomePage(driver);
+
```
Comment (edsonarios) [inline src/main/java/salesforce/ui/pages/LoginPage.java:56]: I think there are plenty of spaces

File: src/main/java/core/DriverManager.java (hunk 2 above)
Comment (danielaSantaCruz) [reply src/main/java/core/DriverManager.java]: Yeah, I already fixed it

File: src/main/java/core/DriverManager.java (hunk 1 above)
Comment (danielaSantaCruz) [reply src/main/java/core/DriverManager.java]: Yeah, I already changed it

File: src/main/java/salesforce/ui/pages/LegalEntityPage.java (hunk 3 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/LegalEntityPage.java]: Already done.

**Review by jpgonzales27 — APPROVED** [review]

File: src/main/java/core/selenium/WebElementAction.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package core.selenium;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+/**
+ * sadsa.
```
Comment (edsonarios) [inline src/main/java/core/selenium/WebElementAction.java]: This comment is not understood

File: src/main/java/core/selenium/WebElementAction.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    private WebDriver driver;
+    private WebDriverWait wait;
+
+    public WebElementAction() {
+        driver = DriverManager.getInstance().getDriver();
+        wait = new WebDriverWait(driver, 30);
+    }
+
+    /**
+     * coment.
```
Comment (edsonarios) [inline src/main/java/core/selenium/WebElementAction.java]: This comment is not understood

File: src/main/java/salesforce/configuration/ConfigurationFile.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+public class ConfigurationFile {
+
+    private static final String CONFIG_FILE_PATH = "config.properties";
+    public static Properties CONFIG_PROPERTIES;
+
+    static {
+        try {
+            CONFIG_PROPERTIES = readFile();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (edsonarios) [inline src/main/java/salesforce/configuration/ConfigurationFile.java]: You should do something with the exception, not just printing it. You could throw it

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import salesforce.ui.pages.LoginPage;
+
+/**
+ * Navigates to an specified url.
+ */
+public class PageTransporter {
+
+    private String baseUrl = getEnvVariables("BASE_URL");
+
+    public void goToUrl(final String url) {
```
Comment (edsonarios) [inline src/main/java/salesforce/ui/PageTransporter.java:22]: Javadoc missing

File: src/main/java/salesforce/ui/pages/LegalEntityPage.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
```
Comment (edsonarios) [inline src/main/java/salesforce/ui/pages/LegalEntityPage.java]: I think, this is a number magic

File: src/main/java/salesforce/ui/pages/LoginPage.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    @FindBy(id = "Login")
+    private WebElement loginBtn;
+
+    @Override
+    protected void waitForPageLoaded() {
+        webElementAction.waitForVisibilityOfElement(loginBtn);
+    }
+
+    public LoginPage setUserName(final String userName) {
```
Comment (edsonarios) [inline src/main/java/salesforce/ui/pages/LoginPage.java:32]: Javadoc missing

File: src/main/java/salesforce/ui/pages/NewLegalEntityPage.java [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    public NewLegalEntityPage() {
+        super();
+    }
+
+    @Override
+    protected void waitForPageLoaded() {
+        wait.until(ExpectedConditions.visibilityOf(saveBtn));
+    }
+
+    public NewLegalEntityPage setNameTxtBox(final String entityName) {
```
Comment (edsonarios) [inline src/main/java/salesforce/ui/pages/NewLegalEntityPage.java:55]: Javadoc missing

File: src/test/java/base/BaseTest.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+    private DriverManager driverManager;
+    protected LoginPage loginPage;
+    protected PageTransporter pageTransporter;
+
+    @BeforeClass
+    public void beforeClass() {
+        pageTransporter = new PageTransporter();
+        driverManager = DriverManager.getInstance();
+//        driver.get(getEnvVariables("BASE_URL"));
```
Comment (edsonarios) [inline src/test/java/base/BaseTest.java]: This comment should be deleted

**Review by edsonarios — APPROVED** [review]

**Pull request merged**